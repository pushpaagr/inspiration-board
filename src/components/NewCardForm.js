import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]



class NewCardForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: "",
      emoji: "",
    };

  }

  resetState = () => {
    this.setState ({
      text: '',
      emoji: '',
    });
  }


  onFormChange = (event) => {
    const text = event.target.name;
    const value = event.target.value;

    const updatedState = {};
    updatedState[text] = value;
    this.setState(updatedState);
  }

  onSubmit = (event) => {
    event.preventDefault();
    const { text, emoji} = this.state;

    if (text === '' || emoji === '') return;

    this.props.addCardCallback(this.state);
    this.resetState();
  }



  render() {
    return(
      <form onSubmit={this.onSubmit} className="new-card-form">

        <div>
          <textarea className="new-card-form__form-textarea" name="text" onChange={this.onFormChange}
            value={this.state.text}></textarea>
        </div>

        <div>
          <label className="new-card-form__form-label" htmlFor="Emoji">Emoji:</label>
          <input name="emoji" placeholder="emoji" onChange={this.onFormChange}
            value={this.state.emoji} />
        </div>

        <input className="new-card-form__form-button" type="submit" name="submit" value="Add a Card" />


      </form>
    );
  }


}

NewCardForm.propTypes = {
  addCardCallback: PropTypes.func.isRequired,
};

export default NewCardForm;
