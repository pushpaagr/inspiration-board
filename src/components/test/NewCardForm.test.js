import React from 'react';
import NewCardForm from '../NewCardForm';
import { shallow } from 'enzyme';

describe('NewCardForm', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = shallow(
      <NewCardForm addCardCallback={() => {} } />
    );

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
