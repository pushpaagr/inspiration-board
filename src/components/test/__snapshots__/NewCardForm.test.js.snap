// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches an existing snapshot 1`] = `
<form
  className="new-card-form"
  onSubmit={[Function]}
>
  <div>
    <textarea
      className="new-card-form__form-textarea"
      name="text"
      onChange={[Function]}
      value=""
    />
  </div>
  <div>
    <label
      className="new-card-form__form-label"
      htmlFor="Emoji"
    >
      Emoji:
    </label>
    <input
      name="emoji"
      onChange={[Function]}
      placeholder="emoji"
      value=""
    />
  </div>
  <input
    className="new-card-form__form-button"
    name="submit"
    type="submit"
    value="Add a Card"
  />
</form>
`;
